### PR TITLE
Issue 1767: security vulnerabilities in 3rd party dependencies

### PR DESCRIPTION
**Pull request by mptap**
Descriptions of the changes in this PR:

Upgraded jline and jackson to remove security vulnerabilities mentioned in #1767 

### Motivation

Remove security vulnerabilities mentioned in #1767 

### Changes

Upgraded jline and jackson 3rd party dependencies

Master Issue: #1767 

> ---
> In order to uphold a high standard for quality for code contributions, Apache BookKeeper runs various precommit
> checks for pull requests. A pull request can only be merged when it passes precommit checks. However running all
> the precommit checks can take a long time, some trivial changes don't need to run all the precommit checks. You
> can check following list to skip the tests that don't need to run for your pull request. Leave them unchecked if
> you are not sure, committers will help you:
>
> - [ ] [skip bookkeeper-server bookie tests]: skip testing `org.apache.bookkeeper.bookie` in bookkeeper-server module.
> - [ ] [skip bookkeeper-server client tests]: skip testing `org.apache.bookkeeper.client` in bookkeeper-server module.
> - [ ] [skip bookkeeper-server replication tests]: skip testing `org.apache.bookkeeper.replication` in bookkeeper-server module.
> - [ ] [skip bookkeeper-server tls tests]: skip testing `org.apache.bookkeeper.tls` in bookkeeper-server module.
> - [ ] [skip bookkeeper-server remaining tests]: skip testing all other tests in bookkeeper-server module.
> - [ ] [skip integration tests]: skip docker based integration tests. if you make java code changes, you shouldn't skip integration tests.
> - [ ] [skip build java8]: skip build on java8. *ONLY* skip this when *ONLY* changing files under documentation under `site`.
> - [ ] [skip build java9]: skip build on java9. *ONLY* skip this when *ONLY* changing files under documentation under `site`.
> ---

> ---
> Be sure to do all of the following to help us incorporate your contribution
> quickly and easily:
>
> If this PR is a BookKeeper Proposal (BP):
>
> - [ ] Make sure the PR title is formatted like:
>     `<BP-#>: Description of bookkeeper proposal`
>     `e.g. BP-1: 64 bits ledger is support`
> - [ ] Attach the master issue link in the description of this PR.
> - [ ] Attach the google doc link if the BP is written in Google Doc.
>
> Otherwise:
> 
> - [x] Make sure the PR title is formatted like:
>     `<Issue #>: Description of pull request`
>     `e.g. Issue 123: Description ...`
> - [x] Make sure tests pass via `mvn clean apache-rat:check install spotbugs:check`.
> - [x] Replace `<Issue #>` in the title with the actual Issue number.
> 
> ---

### CODE REVIEW

**Comment by mptap** [comment]
Thanks @sijie That makes sense, except I'm not sure why we don't want to upgrade the scale version to 2.12 from 2.11? https://mvnrepository.com/artifact/com.fasterxml.jackson.module/jackson-module-scala_2.12/2.9.7

**Comment by sijie** [comment]
@mptap 

so `jackson-module-scala_2.11:jar` is a transitive dependency of ` com.twitter:finagle-core_2.11:jar`. if we want to bump the version to 2.12, we have to bump all scala related dependencies to 2.12. but that would be a separated change. so enforcing it to 2.11 is good enough for this PR.

```
[INFO] org.apache.bookkeeper.stats:twitter-finagle-provider:jar:4.9.0-SNAPSHOT
[INFO] +- org.apache.bookkeeper.stats:bookkeeper-stats-api:jar:4.9.0-SNAPSHOT:compile
[INFO] +- com.twitter:finagle-core_2.11:jar:6.44.0:compile
[INFO] |  +- org.scala-lang:scala-library:jar:2.11.8:compile
[INFO] |  +- com.twitter:finagle-toggle_2.11:jar:6.44.0:compile
[INFO] |  |  +- com.fasterxml.jackson.core:jackson-core:jar:2.8.9:compile
[INFO] |  |  +- com.fasterxml.jackson.core:jackson-databind:jar:2.8.9:compile
[INFO] |  |  |  \- com.fasterxml.jackson.core:jackson-annotations:jar:2.8.9:compile
[INFO] |  |  +- com.fasterxml.jackson.module:jackson-module-scala_2.11:jar:2.8.4:compile
[INFO] |  |  |  \- com.fasterxml.jackson.module:jackson-module-paranamer:jar:2.8.4:compile
[INFO] |  |  |     \- com.thoughtworks.paranamer:paranamer:jar:2.8:compile
[INFO] |  |  \- com.google.guava:guava:jar:21.0:compile
```

**Comment by sijie** [comment]
run bookkeeper-server replication tests

File: bookkeeper-dist/src/main/resources/LICENSE-all.bin.txt [hunk 1]
```diff
@@ -300,11 +300,11 @@ Apache Software License, Version 2.
...
-[1] Source available at https://github.com/FasterXML/jackson-annotations/tree/jackson-annotations-2.8.9
-[2] Source available at https://github.com/FasterXML/jackson-core/tree/jackson-core-2.8.9
-[3] Source available at https://github.com/FasterXML/jackson-databind/tree/jackson-databind-2.8.9
-[4] Source available at https://github.com/FasterXML/jackson-modules-base/tree/jackson-modules-base-2.8.4
-[5] Source available at https://github.com/FasterXML/jackson-module-scala/tree/f9e099
+[1] Source available at https://github.com/FasterXML/jackson-annotations/tree/jackson-annotations-2.9.7
+[2] Source available at https://github.com/FasterXML/jackson-core/tree/jackson-core-2.9.7
+[3] Source available at https://github.com/FasterXML/jackson-databind/tree/jackson-databind-2.9.7
+[4] Source available at https://github.com/FasterXML/jackson-modules-base/tree/jackson-modules-base-2.9.7
+[5] Source available at https://github.com/FasterXML/jackson-module-scala/tree/jackson-module-scala-2.9.7
```
Comment (eolivelli) [inline bookkeeper-dist/src/main/resources/LICENSE-all.bin.txt]: Do we really need this jar?

Comment (sijie) [reply bookkeeper-dist/src/main/resources/LICENSE-all.bin.txt]: this is `all` package, it is including all stats providers. that specific dependency is coming from finagle stats provider, I guess it is required for that provider to be working.

**Comment by sijie** [comment]
run bookkeeper-server bookie tests
run bookkeeper-server replication tests

File: bookkeeper-dist/src/main/resources/LICENSE-all.bin.txt (hunk 1 above)
Comment (eolivelli) [reply bookkeeper-dist/src/main/resources/LICENSE-all.bin.txt]: I see

**Comment by sijie** [comment]
@mptap thank you for your contribution

**Pull request merged**